code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/unit/DefaultDeploymentUnitInstance.java

```diff
@@ -330,7 +331,19 @@ protected ServiceIndex createServiceIndex() {
                     InstanceConstants.FIELD_REQUESTED_IP_ADDRESS);
             String requestedIp = null;
             if (requestedIpObj != null) {
-                requestedIp = requestedIpObj.toString();
+                List<String> availableIPs = new ArrayList<>();
+                if (requestedIpObj instanceof List<?>) {
+                    for (Object item : (List<?>) requestedIpObj) {
+                        availableIPs.add(ObjectUtils.toString(item, null));
+                    }
+                }
+                List<String> usedIPs = context.serviceDao.getUsedIPsFromServiceIndexByServiceID(service);
```

Comment:
Any particular reason we are validating against used IPs used only within the same service? Why not within services of the same account? But even that might not be enough as the requested ip address can be used by standalone container. Also from what I recall, in ip allocation code down below, pool manager would return nothing if the ip address is already allocated in the subnet:

https://github.com/rancher/cattle/blob/b3d7be606152793f615107bdd8ae2017e2e32b62/code/iaas/logic-common/src/main/java/io/cattle/platform/network/impl/NetworkServiceImpl.java#L83

if we prove that the pool manager validates against the duplicates for service requested ips, we can omit adding one extra layer of validation in services code